### ISSUE TITLE
compatibility with the OS-VBS MKII project

### ISSUE THREAD

**Issue by RightRudderLeftStick** [JustEnoughDucks/LibreMiG-S#2]
Hi, just came across this project on reddit, is it still under development?

I wonder if you'd consider compatibility with my project:
https://github.com/RightRudderLeftStick/OpenSource-VeryBigStick

It's CC BY-SA 2.0, and the only real requirement is for the stick to mount on the end of 2020 extrusion directly.

**Comment by JustEnoughDucks** [comment]
Hey, I ended moving and started a full renovation, so it kind of dropped by the side as my 3D printer is unavailable for another year. My final full assembly build is still sitting in a box. I will see if I can restart it soon

I will definitely consider adding it. From what I can see of your project, it is just mechanical right now, correct? The issue I see right now is wire paths for bridging sensor to grip. That is difficult while using aluminum extrusions.

This stick uses the TM interface, so it should work with your stick on your already made extrusion-to-TM adapter, if I am not mistaken. Going to the [mechanical source files: ](https://github.com/JustEnoughDucks/LibreMiG-S/tree/8a1e5ddca9180a5b205c3cc3732c93596d07cc1f/Mechanical/source/assembly) and looking at the TM adapter and TM screw, you should be able to open them in FreeCAD and export it to STEP to fit it with your design in F360.

Maybe the most interesting for you will be when I do an electronics redesign with a mini PCB as the base controller along with a more robust I2C interface. I would be open to working together if you need electronics work done or want to make the systems cross-compatible.

**Comment by RightRudderLeftStick** [comment]
Funny that you mention TM, I am right now working on a freejoy/mmjoy2/arduino breakout board and the stick will interface over the CD4021 connection:

https://www.patreon.com/posts/working-on-pcb-99781394

The sensors will be those TLE5010 kits on aliexpress since they are ready made. ( https://www.aliexpress.com/item/4000549883214.html )

**Comment by JustEnoughDucks** [comment]
Yeah as I got into it, I realized that electrically matching TM's SPI/SSC/shift register interfaces leaves little flexibility because you are limited by chip select wires. I2C, while slower (for HID usecase still faster than USB can poll), allows for modularity and adding or removing more sensors with different stick iterations. 

As far as the TLE5010 kits, since sensors on sticks never are 2 axis like a joystick and don't make full rotations, better accuracy and precision can be achieved with linear hall effect sensors, good mechanical design, and a good ADC. those angular sensors, you might only end up using 10 bits of the 16 bit conversion as an example since you only use a few degrees of its range for things like triggers or twist. Either one you choose, linear or all-in-one angular, a custom solution is going to be the easiest to get working reliably is fully custom (in the end), but the downside is that then you have to sell kits because most people won't want to solder a whole board themselves or have the equipment to do so.

**Comment by RightRudderLeftStick** [comment]
Yeah wanted to keep stuff as much "off the shelf" as possible, thus sticking to the aliexpress TLE5010 kits for now.

btw just dropped a big update for the project with an electrical guide using MMJOY2 and a custom breakout board. I am going to follow up with a bluepill based breakout board next.

https://github.com/RightRudderLeftStick/OpenSource-VeryBigStick
